### PR TITLE
Add Decision Trees Colab

### PR DESCRIPTION
**Pull request by SauravMaheshkar**

### CODE REVIEW

File: colabs/scikit/decision-trees.ipynb [hunk 1]
```diff
@@ -0,0 +1,240 @@
+{
```
Comment (ayulockin) [inline colabs/scikit/decision-trees.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #5.    os.environ["WANDB_API_KEY"] = '...'
</pre><p>Are we using <code>os.environ</code> to ask  user to login in every example? I think using <code>wandb.login()</code>  should be better since it will show the https://wandb.ai/authorize link to get the api token. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/wandb/examples/pull/106/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (ayulockin) [inline colabs/scikit/decision-trees.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    run = wandb.init(project='...', entity='...', config = config)
</pre><p>Should we remove the <code>entity</code>  argument and provide some dummy project name here?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/wandb/examples/pull/106/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (ayulockin) [inline colabs/scikit/decision-trees.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #17.    wandb.sklearn.plot_confusion_matrix(y_test, y_pred, config.labels)
</pre><p>Not sure if we are promoting this api usage. Correct me if I am wrong. Should we use <a href="https://wandb.ai/wandb/plots/reports/Confusion-Matrix--VmlldzozMDg1NTM?galleryTag=visualization" target="_blank">this</a> method instead?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/wandb/examples/pull/106/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (ayulockin) [inline colabs/scikit/decision-trees.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    run = wandb.init(project='...', entity='...', config = config)
</pre><p>Same as above.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/wandb/examples/pull/106/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (ayulockin) [inline colabs/scikit/decision-trees.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #14.    reg = reg.fit(x_train,y_train)
</pre><p>Don't we have any callback for sklearn's decision tree implementation? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/wandb/examples/pull/106/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (SauravMaheshkar) [reply colabs/scikit/decision-trees.ipynb:1]: Good point. I use `os.environ` as it allows for quick prototyping. Will make the changes.

Comment (SauravMaheshkar) [reply colabs/scikit/decision-trees.ipynb:1]: I plan to do this with all the "Fundamental Guides". I'll create another team and put that `entity` afterwards.

Comment (SauravMaheshkar) [reply colabs/scikit/decision-trees.ipynb:1]: Was unaware of this feature 😅. Will run next experiments and then revert.

Comment (SauravMaheshkar) [reply colabs/scikit/decision-trees.ipynb:1]: Huh, which one ? That'd make the report better as well. I can't find any references in the docs.

Comment (ayulockin) [reply colabs/scikit/decision-trees.ipynb:1]: Then you will have to open the project so that anyone can submit a run there.

Comment (ayulockin) [reply colabs/scikit/decision-trees.ipynb:1]: I don't think we have. Currently we only cover XGBoost and LightGBM. Both of them have PRs waiting to be merged. We will soon be supporting CatBoost. Maybe we can sync and I can add a callback for sklearn as well.

Comment (SauravMaheshkar) [reply colabs/scikit/decision-trees.ipynb:1]: Good point, would you suggest anony mode instead ? or should I just add my id ?

Comment (SauravMaheshkar) [reply colabs/scikit/decision-trees.ipynb:1]: Sure sounds good. Till then let's stick to this implementation ? Thoughts ??

Comment (ayulockin) [reply colabs/scikit/decision-trees.ipynb:1]: I don't think we should use anonymous mode here. I think anyone who will use our examples colab will have the desire to create an account of his/her own. How about we just provide a dummy project name without entity. That way when someone runs the colab that person will create a new project in his own entity. 

With anonymous mode there's an extra step of claiming the run if someone is interested. For a first timer it might be confusing. etc.

Comment (ayulockin) [reply colabs/scikit/decision-trees.ipynb:1]: Since I didn't run your colab, are you able to log any metrics? Can you share an example dashboard? As I see it only confusion matrix is logged.

Comment (SauravMaheshkar) [reply colabs/scikit/decision-trees.ipynb:1]: Sure Link to the workspace -> https://wandb.ai/sauravmaheshkar/Decision-Tree

Comment (ayulockin) [reply colabs/scikit/decision-trees.ipynb:1]: * The colab showcase the features we have currently. 
* We ain't logging any metrics since we don't have any callback. This can be a todo for future. For now I think we can merge re: point 1.

**Comment by tcapelle** [comment]
I would love more text in general, explaining what you are doing. If there is no text this should be a .py file.

**Comment by tcapelle** [comment]
I am closing this now, feel free to reopen when you are ready.
